app/controllers/foreman_deployments/api/v2/stacks_controller.rb

```diff
@@ -20,6 +20,18 @@ def create
           process_response @stack.save
         end
 
+        api :PUT, '/stacks/:id/', N_('Update imported stack')
+        param_group :stack, :as => :update
+        param :id, :identifier, :required => true
+        def update
+          if !@stack.configurations.empty? && !params[:stack][:definition].nil?
```

Comment:
I agree. Since there's other priority work to do now, let's go with the refactor issue: http://projects.theforeman.org/issues/11904